e2e/playwright/upload-download-actions/src/tests/version-actions.e2e.ts

```diff
@@ -0,0 +1,209 @@
+/*!
+ * Copyright © 2005-2024 Hyland Software, Inc. and its affiliates. All rights reserved.
+ *
+ * Alfresco Example Content Application
+ *
+ * This file is part of the Alfresco Example Content Application.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail. Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * The Alfresco Example Content Application is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The Alfresco Example Content Application is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * from Hyland Software. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { expect } from '@playwright/test';
+import {
+  ApiClientFactory,
+  Utils,
+  test,
+  TrashcanApi,
+  NodesApi,
+  TEST_FILES,
+  FileActionsApi,
+  SharedLinksApi,
+  FavoritesPageApi,
+  RecentFilesPage,
+  LoginPage,
+  PersonalFilesPage,
+  FavoritesPage,
+  SharedPage,
+  SearchPage
+} from '@alfresco/playwright-shared';
+
+test.describe('Version actions', () => {
+  let trashcanApi: TrashcanApi;
+  let nodesApi: NodesApi;
+  let fileActionsApi: FileActionsApi;
+  let sharedLinksApi: SharedLinksApi;
+  let favoritesApi: FavoritesPageApi;
+  const random = Utils.random();
+  const filesToUpload = [TEST_FILES.PDF, TEST_FILES.DOCX];
+  const filenameBeforeUpdate = `${filesToUpload[0].name}-${random}`;
+  const filenameAfterUpdate = `${filesToUpload[1].name}-${random}`;
+  const username = `user-${random}`;
+  const parentFolder = `parent-${random}`;
+  let parentFolderId: string;
+  let fileId: string;
+
+  async function viewFirstFileVersion(loginPage: LoginPage, page: PersonalFilesPage | RecentFilesPage | FavoritesPage | SharedPage | SearchPage) {
```

Comment:
here function serves two things not aline with its name, like login and navigate ,
as we separate describe for personal file and search we can have login and navigate to home/search page operations in before each and view file version as a common function